backend/unit-tests/focs-2-3-tests.ml

```diff
@@ -0,0 +1,148 @@
+(*helper functions*)
+
+let rec intsFromTo x y = 
+if x>y then [] else x::(intsFromTo (x+1) y);;
+
+let intsTo n = 
+let rec intsFrom m =
+if m>n then [] else m::(intsFrom (m+1))
+in
+intsFrom 1 ;;
+
+let removeDupes xs = 
+	let rec aux y xs = match xs with 
+	| [] -> y
+	| x::xs -> aux (x::y) (List.filter (fun i-> i!=x) xs) in
+aux [] xs;; 
+
+let rec rndPerm list =
+    let rec extract acc n = function
+      | [] -> []
+      | x ::xs -> if n = 0 then (x, acc @ xs) else extract (x::acc) (n-1) xs
+    in
+    let extract_rand list len =
+      extract [] (Random.int len) list
+    in
+    let rec aux acc list len =
+      if len = 0 then acc else
+        let picked, rest = extract_rand list len in
+        aux (picked :: acc) rest (len-1)
+    in
+    aux [] list (List.length list);;
+
+(*intersect*)
+
+let testOnEmptyLists = assert_equal [] (Submission.intersect [] []);;
```

Comment:
These unit test names are very undescriptive. Please could we say exactly what property we're testing?

Each unit test should test exactly one property of the function, and it should be clear in what way the function is wrong if the unit test fails.

As an example, this would be `testIntersectionOfEmptyListsIsTheEmptyList`.